source/fundamentals/authentication.txt

```diff
@@ -142,6 +143,157 @@ string as follow:
 
    To learn more on specifying the default mechanism, see :ref:`csharp-authentication-default`.
 
+.. _csharp-mongodb-aws:
+
+MONGODB-AWS
+~~~~~~~~~~~
+
+.. note::
+
+   The ``MONGODB-AWS`` authentication mechanism is available only for
+   MongoDB deployments on MongoDB Atlas.
+
+The ``MONGODB-AWS`` authentication mechanism uses your Amazon Web Services
+Identity and Access Management (AWS IAM) credentials to authenticate your
+user. You can either specify your credentials directly on a ``MongoClientSettings`` object 
+or instruct the driver to retrieve them automatically from an external source.
+
+The following sections contain code examples that use the following placeholders:
+
+- ``<awsKeyId>`` - value of your AWS access key ID
+- ``<awsSecretKey>`` - value of your AWS secret access key
+- ``<atlasUri>`` - hostname and port of your MongoDB Atlas deployment
+- ``<awsSessionToken>`` - value of your AWS session token
+
+.. tip::
+
+   To learn more about configuring MongoDB Atlas with AWS IAM, see the
+   :atlas:`Set Up Passwordless Authentication with AWS IAM Roles </security/passwordless-authentication/#set-up-passwordless-authentication-with-aws-iam-roles>` guide.
+
+Specify Your Credentials in MongoClientSettings
++++++++++++++++++++++++++++++++++++++++++++++++
+
+You can supply your AWS IAM credentials on a ``MongoClientSettings`` object either by 
+using a ``MongoCredential`` object or as part of the connection string. Select the 
+:guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax for specifying your credentials: 
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: mongodb-aws-mongoclientsettings-connection-string
+      
+      .. code-block:: csharp
+
+         var connectionString = "mongodb+srv://<awsKeyId>:<awsSecretKey>@<atlasUri>?authSource=$external&authMechanism=MONGODB-AWS&authMechanismProperties=AWS_SESSION_TOKEN:<awsSessionToken>";
+         var mongoClientSettings = MongoClientSettings.FromConnectionString(connectionString);
+         var client = new MongoClient(mongoClientSettings);
+
+      .. note:: Omitting AWS Session Token
+
+         If you aren't using an AWS session token, omit the ``authMechanismProperties``
+         parameter.
+
+   .. tab:: MongoCredential
+      :tabid: mongodb-aws-mongoclientsettings-mongo-credential
+
+      .. code-block:: csharp
+
+         var mongoClientSettings = MongoClientSettings.FromConnectionString("mongodb+srv://<atlasUri>");
+         mongoClientSettings.Credential = new MongoCredential("MONGODB-AWS", new MongoExternalIdentity("<awsKeyId>"), new PasswordEvidence("<awsSecretKey>"));
+             .WithMechanismProperty("AWS_SESSION_TOKEN", "<awsSessionToken>");
+         var client = new MongoClient(mongoClientSettings);
+
+      .. note:: Omitting AWS Session Token
+
+         If you aren't using an AWS session token, omit the ``WithMechanismProperty()``
+         method call.
+
+Retrieve Credentials Automatically
```

Comment:
Good question for the DBX team.

Hey @DmitryLukyanov, does it make sense for the C# docs to include a section on the AWS SDK in the same way that the Java docs (linked above) do? This would include adding the SDK to your project, supplying credentials through a provider chain, and any other relevant topics.

Reply:
Thanks!

@ccho-mongodb, given what Dima said, I don't see the need for a ticket here. Do you agree?

Reply:
Ok, fine with not including a provider chain. I did find some info on it though, in case it was helpful to some users:
https://www.stevejgordon.co.uk/credential-loading-and-the-aws-sdk-for-dotnet-deep-dive

Reply:
It's not official documentation, so I think we can't use it in our docs. Also, we use only `FallbackCredentialsFactory`, the rest of description is unrelated to us